File: frontend/bins/src/views/landmarks/js/Petition.js
import User from "@/js/auth/User.js";

class Petition {
    constructor(id, userId, lat, lon, landmarkInfo, accepted, creation, edition) {
        this.id = id;
        this.userId = userId;
        this.lat = lat;
        this.lon = lon;
        this.landmarkInfo = landmarkInfo;
        this.accepted = accepted;
        this.creation = creation;
        this.edition = edition;
    }
}

export default Petition

File: backend/streetlight/src/main/java/app/caquita/landmark/forum/petitions/Petition.java
package app.caquita.landmark.forum.petitions;

import app.caquita.landmark.LandmarkType;
import org.jdbi.v3.json.Json;

import java.sql.Timestamp;
import java.util.HashMap;

public record Petition(
        long id, String name, String description, long userId, LandmarkType type,
        double lat, double lon,
        @Json HashMap<String, Boolean> info,
        Status status, Timestamp creation, Timestamp edition
) {
    public enum Status {
        PENDING,
        ACCEPTED,
        REJECTED
    }
}